pkg/asset/machines/master.go

```diff
@@ -539,6 +539,14 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		}
 		machineConfigs = append(machineConfigs, ignMultipath)
 	}
+	if ic.Platform.Name() == openstacktypes.Name && len(installConfig.Config.ServiceNetwork) == 2 {
```

Comment:
I'm surprised here that we do not check the ip version of strings in `installConfig.Config.ServiceNetwork`; and if there is a ipv6 match we enter in the condition.
What if the `installConfig.Config.ServiceNetwork` has two IPv4 service networks?

Reply:
I followed the same rationale from here https://github.com/openshift/installer/blob/master/pkg/asset/ignition/bootstrap/common.go#L291-L297

Also only one v4 and v6 CIDR max is allowed, according to this [code fragment](https://github.com/openshift/cluster-network-operator/blob/master/pkg/network/cluster_config.go#L50-L52).

Reply:
Indeed, https://github.com/openshift/cluster-network-operator/blob/fc3e0e25b4cfa43e14122bdcdd6d7f2585017d75/pkg/network/cluster_config.go#L46-L52 validates it in CNO.
I guess this is fine, as we probably don't want to repeat the same validation logic in the installer; in case it changes in CNO at some point.

As someone who reads this code for the first time, I would have appreciated a comment explain that ^ with a permalink to the CNO code eventually.
So whoever comes back to it knows why we "only" validate that ServiceNetwork has 2 strings and the rest is done via CNO.